nexus/tests/integration_tests/authz.rs

```diff
@@ -0,0 +1,439 @@
+        NexusRequest::object_get(client, &"/session/me/sshkeys")
+            .authn_as(AuthnMode::SiloUser(user2))
+            .execute()
+            .await
+            .expect("failed to make GET request")
+            .parsed_body()
+            .unwrap();
+
+    assert!(user2_keys.items.is_empty());
+}
+
+// Test that an authenticated, unprivileged user can list and read global images
+#[nexus_test]
+async fn test_global_image_read_for_unpriv(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let client = &cptestctx.external_client;
+    let nexus = &cptestctx.server.apictx.nexus;
+
+    // Create a silo with an unprivileged user
+    let silo =
+        create_silo(&client, "authz", true, shared::UserProvisionType::Fixed)
+            .await;
+
+    let new_silo_user_id = Uuid::new_v4();
+    nexus
+        .silo_user_create(silo.identity.id, new_silo_user_id, "unpriv".into())
+        .await
+        .unwrap();
+
+    // Create a global image using AuthnMode::PrivilegedUser
+    let server = ServerBuilder::new().run().unwrap();
+    server.expect(
+        Expectation::matching(request::method_path("HEAD", "/image.raw"))
+            .times(1..)
+            .respond_with(
+                status_code(200).append_header(
+                    "Content-Length",
+                    format!("{}", 4096 * 1000),
+                ),
+            ),
+    );
+
+    let image_create_params = params::GlobalImageCreate {
+        identity: IdentityMetadataCreateParams {
+            name: "alpine-edge".parse().unwrap(),
+            description: String::from(
+                "you can boot any image, as long as it's alpine",
+            ),
+        },
+        source: params::ImageSource::Url {
+            url: server.url("/image.raw").to_string(),
+        },
+        distribution: params::Distribution {
+            name: "alpine".parse().unwrap(),
+            version: "edge".into(),
+        },
+        block_size: params::BlockSize::try_from(512).unwrap(),
+    };
+
+    NexusRequest::objects_post(client, "/images", &image_create_params)
+        .authn_as(AuthnMode::PrivilegedUser)
+        .execute()
+        .await
+        .unwrap();
+
+    // The unprivileged user:
+
+    // - can list global images
+    let _images: ResultsPage<views::GlobalImage> =
+        NexusRequest::object_get(client, &"/images")
+            .authn_as(AuthnMode::SiloUser(new_silo_user_id))
+            .execute()
+            .await
+            .expect("failed to make GET request")
+            .parsed_body()
+            .unwrap();
+
+    // - can read a global image
+    let _image: views::GlobalImage =
+        NexusRequest::object_get(client, &"/images/alpine-edge")
+            .authn_as(AuthnMode::SiloUser(new_silo_user_id))
+            .execute()
+            .await
+            .expect("failed to make GET request")
+            .parsed_body()
+            .unwrap();
+
+    // - cannot create a global image - should get 403
```

Comment:
I think this should be covered by the unauthorized test (commented there that we should still test ReadOnly endpoints)

Reply:
Addressed in #1399.